subprojects/model-core/src/main/java/org/gradle/api/internal/provider/DefaultProperty.java

```diff
@@ -112,8 +112,11 @@ public void set(Provider<? extends T> provider) {
     }
 
     @Override
-    public Property<T> convention(Provider<? extends T> valueProvider) {
-        setConvention(Providers.internal(valueProvider).asSupplier(getValidationDisplayName(), type, sanitizer));
+    public Property<T> convention(Provider<? extends T> provider) {
+        if (provider == null) {
+            throw new IllegalArgumentException("Cannot set the convention of a property using a null provider.");
```

Comment:
At my previous job, we would always use Guava's `com.google.common.base.Preconditions` helpers for checks like this, to make it more obvious when an if check was part of the logic in body of the method vs. when it was input validation.

`Preconditions.checkArgument(provider != null, "Cannot set the convention of a property using a null provider.");`

I see only a little usage of that class in Gradle code.  Is it something to be avoided?

Reply:
I don't know. Is quite nice, will put it in there (it's an internal class anyways), let's see if anything creaks during building.